integrationtests/src/test/java/io/kroxylicious/proxy/KroxyConfigBuilder.java

```diff
@@ -57,6 +66,11 @@ public KroxyConfigBuilder withKeyStoreConfig(String keystoreFile, String keyPass
         return this;
     }
 
+    public KroxyConfigBuilder withAdminHttpPrometheusEndpoint() {
```

Comment:
Not wild about the combined nature of this leaking into the API. 

My gut would be just call the method `withPromehteusEndpoint`, and then do a lazy init of `adminHttp` if required, otherwise just add the new endpoint.

Reply:
Because the record fields are immutable, the `adminHttp` lazy init and addition of the endpoint code currently look identical (you have to re-assign to `adminHttp` and new it all up). That would change once multiple endpoints are supported, you'd want to copy the endpoints and update your relevant endpoint config. Will change the name though to present a better API.